### PR TITLE
Add squid balances

### PR DESCRIPTION
**Pull request by isSerge**
Adding our first squid for block explorer - Balances. Some stuff here is auto-generated (models, interfaces, etc.). The main part is `processor.ts`. It tracks events from `pallet-balances` and updates database

### CODE REVIEW

**Review by jfrank-summit — COMMENTED** [review]
Given the squid is going to hold more than balances I would rename the folder to something more general like `squid-blockexplorer`

File: squid-blockexplorer/src/types/events.ts [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  }
+
+  /**
+   * Some amount was withdrawn from the account (e.g. for transaction fees).
+   */
+  get asV3(): {who: Uint8Array, amount: bigint} {
+    assert(this.isV3)
+    return this._chain.decodeEvent(this.event)
+  }
+}
```
Comment (achiurizo) [inline squid-blockexplorer/src/types/events.ts:297]: should we do some class inheritance here? The distinct changes are the `event.name` and `EventHash`.

File: squid-blockexplorer/src/types/storage.ts [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    return this._chain.queryStorage(this.blockHash, 'System', 'Account')
+  }
+
+  /**
+   * Checks whether the storage item is defined for the current chain version.
+   */
+  get isExists(): boolean {
+    return this._chain.getStorageItemTypeHash('System', 'Account') != null
+  }
+}
```
Comment (achiurizo) [inline squid-blockexplorer/src/types/storage.ts:266]: similar comment to `Events`, maybe some inheritance?

File: squid-blockexplorer/.gitignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/node_modules
+/lib
+
+/**Versions.json
+/**Versions.jsonl
+
+# IDE files
+/.idea
```
Comment (achiurizo) [inline squid-blockexplorer/.gitignore:8]: nitpick: Maybe this should be [in a local `gitignore`](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer)?

File: squid-blockexplorer/.gitignore [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/node_modules
+/lib
+
+/**Versions.json
+/**Versions.jsonl
```
Comment (achiurizo) [inline squid-blockexplorer/.gitignore:5]: what's `.jsonl`?

File: squid-blockexplorer/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    image: postgres:14
+    environment:
+      POSTGRES_DB: squid
+      POSTGRES_PASSWORD: squid
+    ports:
+      - "${DB_PORT}:5432"
+      # command: ["postgres", "-c", "log_statement=all"]
+#    volumes:
+#      - ./data/db:/var/lib/postgresql/data
+
```
[the remark this answers is not in the document]
Comment (isSerge) [reply squid-blockexplorer/docker-compose.yml:15]: that's from the original `subsquid-template` repo, but I agree

File: squid-blockexplorer/src/types/events.ts (hunk 1 above)
Comment (isSerge) [reply squid-blockexplorer/src/types/events.ts:297]: this is autogenerated

File: squid-blockexplorer/.gitignore (hunk 4 above)
Comment (isSerge) [reply squid-blockexplorer/.gitignore:5]: not sure, this is also from the original template [code](https://github.com/subsquid/squid-template/blob/main/.gitignore)

File: squid-blockexplorer/src/types/events.ts (hunk 1 above)
Comment (jfrank-summit) [reply squid-blockexplorer/src/types/events.ts:297]: i do believe this is autogenerated with `typegen.json`.

File: squid-blockexplorer/src/types/storage.ts (hunk 2 above)
Comment (jfrank-summit) [reply squid-blockexplorer/src/types/storage.ts:266]: I believe everything under the `types` folder is autogenerated.

File: squid-blockexplorer/src/types/events.ts (hunk 1 above)
Comment (isSerge) [reply squid-blockexplorer/src/types/events.ts:297]: yes, I just tried and can confirm

**Review by achiurizo — APPROVED** [review]

**Review by jfrank-summit — APPROVED** [review]
I think once we have better defined requirements how we approach `Account` will have to change but this is a good start to getting the repo setup.

**Pull request merged**